solana/gatekeeper-lib/src/service/GatekeeperService.ts

```diff
@@ -26,6 +29,27 @@ export type GatekeeperConfig = {
   defaultExpirySeconds?: number;
 };
 
+/**
+ * Create a new GatewayToken from an existing, setting the state to the given option state and optionally updating the expiry
+ * {GatewayToken} existingToken
+ * options.state: {State}
+ * options.expiry: {number}
+ * returns {GatewayToken}
+ */
+const tokenFromExisting = (
```

Comment:
would this be better off as a function on GatewayToken?

```
const newGatewayToken = gatewayToken.update({ state, expiry })
```

Reply:
gateway token properties are currently read-only on constructor so I'd have to change that but it's not a bad idea

Reply:
updated to add the 'update' function